redirect/client.js

```diff
@@ -0,0 +1,52 @@
+(function() {
+
+    function apply(context, template) {
+
+        /**
+         * Prevent redirect from occurring while in either the Template Editor or Campaign Editor.
+         */
+        if ((window.frameElement || {}).id === "siteEditorFrame"
+            || !context.targetPage.includes(window.location.hostname + window.location.pathname.replace(/\/$/gim, ""))) {
+            return;
+        }
+
+        return new Promise((resolve, reject) => {
+            if ((context.targetPage && context.urlForRedirect)
+                && window.location.href !== context.urlForRedirect) {
+
+                Evergage.cashDom("body").css("visibility", "hidden");
+
+                Evergage.sendStat({
+                    campaignStats: [
+                        {
+                            control: false,
+                            experienceId: context.experience,
+                            stat: "Impression"
+                        }
+                    ]
+                });
+
+                context.paramsForRedirect = (context.maintainQueryParams && window.location.href.match(/\?.*/gim))
+                    ? window.location.href.match(/\?.*/gim)[0]
```

Comment:
You can simply use window.location.search to grab all query parameters. 
```suggestion
                    ? window.location.search
```

Reply:
This doesn't always work, such as the following scenario: https://github.com/evergage/evergage-global-templates/pull/20#issuecomment-718872467